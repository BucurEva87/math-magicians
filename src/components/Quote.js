import React from 'react';

import style from '../styles/Quote.module.scss';

const Quote = () => (
  <section className={style['container-section']} data-testid="quote">
    <article data-testid="quote-article">
      <blockquote data-testid="quote-blockquote">
        <p data-testid="quote-p">
          <em data-testid="quote-em">
            Mathematics is not about numbers, equations, computations or algorithms. It is about
            understanding.
          </em>
        </p>
        <footer data-testid="quote-footer">William Paul Thurstone</footer>
      </blockquote>
    </article>
  </section>
);

export default Quote;
