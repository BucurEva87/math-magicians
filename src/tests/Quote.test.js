import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Quote from '../components/Quote';

let getByTestId;

beforeEach(() => {
  const component = render(<Quote />);
  getByTestId = component.getByTestId;
});

describe('Quote component', () => {
  test('render on screen', () => {
    expect(getByTestId('quote')).toBeInTheDocument();
  });

  test('renders article element on the page', () => {
    expect(getByTestId('quote-article')).toBeInTheDocument();
  });

  test('renders blockquote element on the page', () => {
    expect(getByTestId('quote-blockquote')).toBeInTheDocument();
  });

  test('renders paragraph element on the page', () => {
    expect(getByTestId('quote-p')).toBeInTheDocument();
  });

  test('renders em element on the page', () => {
    expect(getByTestId('quote-em')).toBeInTheDocument();
  });

  test('renders footer element on the page', () => {
    expect(getByTestId('quote-footer')).toBeInTheDocument();
  });

  test('renders em element containg a quote', () => {
    expect(getByTestId('quote-em')).toHaveTextContent(
      'Mathematics is not about numbers, equations, computations or algorithms. It is about understanding.',
    );
  });

  test('renders footer element containg the author', () => {
    expect(getByTestId('quote-footer')).toHaveTextContent('William Paul Thurstone');
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
