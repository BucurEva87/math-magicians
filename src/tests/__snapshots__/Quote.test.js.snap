// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Component matches SnapShot 1`] = `
<section
  className="container-section"
  data-testid="quote"
>
  <article
    data-testid="quote-article"
  >
    <blockquote
      data-testid="quote-blockquote"
    >
      <p
        data-testid="quote-p"
      >
        <em
          data-testid="quote-em"
        >
          Mathematics is not about numbers, equations, computations or algorithms. It is about understanding.
        </em>
      </p>
      <footer
        data-testid="quote-footer"
      >
        William Paul Thurstone
      </footer>
    </blockquote>
  </article>
</section>
`;
